parlai/core/teachers.py

```diff
@@ -2377,32 +2416,46 @@ def _enqueue_request(self):
         """
         Queue a request for loading to the data loader.
         """
-        self.data_loader.request_load(self.receive_data, self.get_chunk, ())
+        if self.threading:
+            self.data_loader.request_load(self.receive_data, self.get_chunk, ())
+        else:
+            self.receive_data(self.get_chunk(), direct_result=True)
 
-    def receive_data(self, future):
+    def receive_data(self, future, direct_result=False):
         """
         Loads data.
 
         Load data into self.samples until buffersize is reached.
         """
-        chunk_output, chunk_reset_cnt = future.result()
+        output = future if direct_result else future.result()
```

Comment:
nit: `direct_result` is a bit ambiguous --> if true, we return the `future`, not the `future.result()`, which is somewhat counterintuitive?

Reply:
Okay, refactored to be more clear I hope